### ISSUE TITLE
App cache size is permanently increased after every publication opening

### ISSUE THREAD

**Issue by mihai-wolfpack** [readium/kotlin-toolkit#12]
### What happened?

After opening any publication (epub/pdf) the application cache size is increased with the respective publication size. The issue is that after closing the reader or the app itself, the cache size is not cleared and if opening the app and the same publication that was cached before, instead of using the cached publication, it again proceeds to cache the publication, resulting in constantly increasing app cache size. 

### Expected behavior

After closing an opened publication, since we do not implement local persistance of the previously opened publications, the cached publication should be cleared. 

### How to reproduce?

In our application we are using the EpubNavigatorFragment/PdfNavigatorFragment for opening the publication.

```childFragmentManager.fragmentFactory =
            PdfNavigatorFragment.createFactory(
                viewModel.publication,
                initialLocator,
                this@PDFReaderFragment
            )
```

We do this after acquiring the publication via a license that we fetch from our backend service, and then adding it to the presentation server as specified in the documentation. We are using the r2-test-app for reference and guidance. Below is the code responsible for adding the publication to the presentation server.

```
private suspend fun acquirePublication(license: ByteArray) {
        _status.value = "Decrypting"

        lcpService.acquirePublication(license) {
            _status.value = "Progress: ${(it * PERCENT_MULTIPLIER).toInt()}%"
        }.onSuccess {
            _status.value = "Decrypted successfully"
            openPublication(it)
        }.onFailure {
            Log.e("ReaderViewModel", it.stackTraceToString())
            clearPublicationDataAndBack()
            _status.value = it.localizedMessage
        }
    }

    private suspend fun openPublication(acquiredPub: LcpService.AcquiredPublication) {
        _status.value = "Opening..."

        val mediaType =
            MediaType.of(fileExtension = File(acquiredPub.suggestedFilename).extension)
        val asset = FileAsset(acquiredPub.localFile, mediaType)

        streamer.open(
            asset,
            allowUserInteraction = false,
            credentials = getCredentials()
        ).onFailure {
            _status.value = it.localizedMessage
            clearPublicationDataAndBack()
        }.onSuccess {
            _status.value = "Opened successfully"
            addToPresentationServer(it)
        }
    }

    private fun addToPresentationServer(publication: Publication) {
        if (publication.isRestricted)
            _status.value = publication.protectionError?.localizedMessage ?: "Unknown error"

        _status.value = "Uploading to publication server"

        val userProperties =
            context.filesDir.path + "/" + Injectable.Style.rawValue + "/UserProperties.json"
        val publicationLocalURL =
            server.addPublication(publication, userPropertiesFile = File(userProperties))

        _publicationLiveData.value = publication
        this.publicationURL = publicationLocalURL
        openEpubReader()
    }
```

In order to observe this behavior, we open the App Info in the phone settings page in order to asses the size of the cache for the app. We have measured the cache size before opening the publication, which gave us the baseline, then again after the navigator fragment finished decrypting and loading the publication and we are actually able to navigate through the pages. Discussing with our backend, we found that the increase in the cache size is roughly the size of the publication. After closing the application we measured the cache size again in the settings page and it was observed that the cache chunk the size of the publication was not cleared. After reproducing the steps above, the cache was yet again increased with the size of the publication.

To summarize the steps to reproduce:
1. Open App Info page to check cache size
2. Open publication inside the application 
3. Observe increase in cache 
4. Close application 
5. Observe the cache is not cleared 
6. Repeat steps (1 - 4)
7. Observe that cache increased with size of publication again

Please not that this is also reproduced on iOS builds using the same readium libraries

### Environment

* Readium version: 2.1.0

#### Development environment

* OS: macOS 11.2
* IDE: Android Studio Arctic Fox 2020.3.1 Patch 2

#### Testing device

* Android version: 10
* Model: OnePlus 5T

**Comment by mihai-wolfpack** [comment]
Yes, we had tested with a ~70 Mb PDF file as well.

**Comment by mickael-menu** [comment]
I took a closer look at your issue and I think it comes from a misunderstanding of the LCP service API.

`LcpService.acquirePublication()` downloads the file in a temporary folder (which is probably listed as Cache in the App Info), but you are responsible for managing the file once downloaded.

Usually you would move the file to a permanent user bookshelf folder. But if you really want to get rid of the PDF file after closing the reader, you need to do it explicitly in your app.

**Comment by mihai-wolfpack** [comment]
I will have a look at this today and will come back with an update as soon as i finish investigating :D 
Thank you for the suggestion!
